### PR TITLE
Removed several unused variables and merged double if statements

### PR DESCRIPTION
**Pull request by MeteorTheLizard**

### CODE REVIEW

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/duplicator.lua [hunk 1]
```diff
@@ -60,7 +60,7 @@ function TOOL:LeftClick( trace )
 
 	DisablePropCreateEffect = true
 
-		local Ents, Constraints = duplicator.Paste( self:GetOwner(), dupe.Entities, dupe.Constraints )
+		local Ents, _ = duplicator.Paste( self:GetOwner(), dupe.Entities, dupe.Constraints )
```
Comment (Kefta) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/duplicator.lua]: The _ can just be removed.

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua [hunk 2]
```diff
@@ -28,7 +28,7 @@ local function ConvertRelativeToEyesAttachment( ent, pos )
 	local attachment = ent:GetAttachment( eyeattachment )
 	if ( !attachment ) then return end
 
-	local LocalPos, LocalAng = WorldToLocal( pos, Angle( 0, 0, 0 ), attachment.Pos, attachment.Ang )
+	local LocalPos, _ = WorldToLocal( pos, Angle( 0, 0, 0 ), attachment.Pos, attachment.Ang )
```
Comment (Kefta) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua]: Same here.

File: garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentheader.lua [hunk 3]
```diff
@@ -27,7 +27,7 @@ end
 
 function PANEL:SizeToContents()
 
-	local w, h = self:GetContentSize()
+	local w, _ = self:GetContentSize()
```
Comment (Kefta) [inline garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentheader.lua]: And here.

Comment (thegrb93) [reply garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentheader.lua]: I think they should remain untouched. Changing it lessens readability.

**Comment by ExtReMLapin** [comment]
FYI, double if condition and `if` + `and` compiles to exactly the same thing

File: garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentheader.lua (hunk 3 above)
Comment (Kefta) [reply garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentheader.lua]: I disagree - if they aren't used, they shouldn't be defined.

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua [hunk 4]
```diff
@@ -28,7 +28,7 @@ local function ConvertRelativeToEyesAttachment( ent, pos )
 	local attachment = ent:GetAttachment( eyeattachment )
 	if ( !attachment ) then return end
 
-	local LocalPos, _ = WorldToLocal( pos, Angle( 0, 0, 0 ), attachment.Pos, attachment.Ang )
+	local LocalPos = WorldToLocal( pos, Angle( 0, 0, 0 ), attachment.Pos, attachment.Ang )
```
Comment (ExtReMLapin) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua]: ree don't recreate angle

Comment (MeteorTheLizard) [reply garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua]: I'm not sure how I would do that.

Comment (thegrb93) [reply garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua]: I don't know what he means either, but I'm assuming he means replace Angle(0,0,0) with angle_zero.

Comment (ExtReMLapin) [reply garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/eyeposer.lua]: yes

**Comment by robotboy655** [comment]
Is there any actual point/benefit in removing the unused variable from multiple argument functions? 
( `local w, h = self:GetContentSize()` )

Also what's up with the inconsistency between outright removing certain unused vars and commenting out the others?

**Comment by Kefta** [comment]
It prevents a local slot from being created, which does save a few cycles.

**Comment by MeteorTheLizard** [comment]
> 
> 
> Is there any actual point/benefit in removing the unused variable from multiple argument functions?
> ( `local w, h = self:GetContentSize()` )
> 
> Also what's up with the inconsistency between outright removing certain unused vars and commenting out the others?

Some were only commented out as they were used by parts of code that is commented out. This is to prevent confusion and errors if someone decides to re-use the code that was commented out in the future.

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua [hunk 5]
```diff
@@ -58,13 +58,14 @@ function TOOL:LeftClick( trace )
...
-		local Norm1, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
+		--local Norm1, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
+		local _, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
 		local LPos1, LPos2 = self:GetLocalPos( 1 ), self:GetLocalPos( 2 )
 		local Phys1 = self:GetPhys( 1 )
 		local WPos2 = self:GetPos( 2 )
 
 		-- Note: To keep stuff ragdoll friendly try to treat things as physics objects rather than entities
-		local Ang1, Ang2 = Norm1:Angle(), ( -Norm2 ):Angle()
+		--local Ang1, Ang2 = Norm1:Angle(), ( -Norm2 ):Angle()
```
Comment (robotboy655) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua:68]:
```suggestion
		local Ang1, Ang2 = Norm1:Angle(), ( -Norm2 ):Angle()

```
These variables are used right below

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua [hunk 6]
```diff
@@ -58,13 +58,14 @@ function TOOL:LeftClick( trace )
 
 		local Ent1, Ent2 = self:GetEnt( 1 ), self:GetEnt( 2 )
 		local Bone1, Bone2 = self:GetBone( 1 ), self:GetBone( 2 )
-		local Norm1, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
+		--local Norm1, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
+		local _, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
```
Comment (robotboy655) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua:62]:
```suggestion
		local Norm1, Norm2 = self:GetNormal( 1 ), self:GetNormal( 2 )
```
You did this in the wrong place, you probably wanted to do this in TOOL:RightClick

File: garrysmod/gamemodes/sandbox/entities/entities/gmod_winch_controller.lua [hunk 7]
```diff
@@ -88,7 +86,7 @@ function ENT:Think()
 
 		if ( per == 0 ) then return true end
 
-		local spos = ( math.sin( ( self.ctime * math.pi * per ) ) + 1 ) * ( amp / 2 )
+		local spos = ( math.sin(  self.ctime * math.pi * per  ) + 1 ) * ( amp / 2 )
```
Comment (robotboy655) [inline garrysmod/gamemodes/sandbox/entities/entities/gmod_winch_controller.lua:89]:
```suggestion
		local spos = ( math.sin( self.ctime * math.pi * per ) + 1 ) * ( amp / 2 )

```
Remove the double spaces

File: garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua [hunk 8]
```diff
@@ -147,12 +148,10 @@ function TOOL:RightClick( trace )
 
 		-- Note: To keep stuff ragdoll friendly try to treat things as physics objects rather than entities
 		local Ang1, Ang2 = Norm1:Angle(), ( -Norm2 ):Angle()
```
Comment (robotboy655) [inline garrysmod/gamemodes/sandbox/entities/weapons/gmod_tool/stools/axis.lua:150]:
```suggestion
		--local Ang1, Ang2 = Norm1:Angle(), ( -Norm2 ):Angle()

```
this is what you wanted to comment out

File: garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentsidebartoolbox.lua [hunk 9]
```diff
@@ -30,7 +30,6 @@ function PANEL:Init()
 	Button:SetSize( 20, 20 )
 	Button:SetCursor( "sizeall" )
 	Button:SetTooltip( "#spawnmenu.header_tooltip" )
-	local slot = Button:Droppable( "SandboxContentPanel" )
```
Comment (robotboy655) [inline garrysmod/gamemodes/sandbox/gamemode/spawnmenu/creationmenu/content/contentsidebartoolbox.lua:33]:
```suggestion
	Button:Droppable( "SandboxContentPanel" )

```

While the variable is unused, `Button:Droppable` is actually necessary for the H1 header thing to be drag'n'droppable.

**Comment by robotboy655** [comment]
Thanks, I will look through this after the game update next week.

**Pull request merged**